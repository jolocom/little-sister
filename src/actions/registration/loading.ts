import {AnyAction} from 'redux'

export const setLoadingMsg = (loadingMsg: string) => {
  return {
    type: 'SET_LOADING_MSG',
    value: loadingMsg
  }
}

export const startLoading = (loading: boolean) => {
  return {
    type: 'START_LOADING',
    value: loading
  }
}

export const finishLoading = (loading: boolean) => {
  return {
    type: 'FINISH_LOADING',
    value: loading
  }
}

export const generateAndEncryptKeyPairs = () => {
  return (dispatch: (actions: AnyAction) => void, {backendMiddleware}) => {
    const randomString = '13912643311766764847120568039921' // TODO: grab from the state
    const password = 'Password1' // TODO: grab from the state

    dispatch(setLoadingMsg('Generating keys'))

    const {
      didDocument,
      mnemonic,
      masterKeyWIF,
      genericSigningKeyWIF,
      ethereumKeyWIF
    } = backendMiddleware.jolocomLib.identity.create(randomString)

    dispatch(setLoadingMsg('Fueling with Ether'))
  }
}
