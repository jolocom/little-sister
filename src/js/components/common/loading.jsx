import React from 'react'
import Radium from 'radium'
import CircularProgress from 'material-ui/CircularProgress'

const Loading = ({style, ...props}) => {
  return (
    <div style={Object.assign({}, styles.container, style)}>
      <CircularProgress {...props} />
    </div>
  )
}

Loading.propTypes = {
  style: React.PropTypes.object
}

const styles = {
  container: {
    height: '100%',
    width: '100%',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center'
  }
}

export default Radium(Loading)
