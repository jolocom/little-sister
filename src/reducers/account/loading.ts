import { AnyAction } from 'redux'
import Immutable from 'immutable'
import { LoadingState } from 'src/reducers/account/'

const initialState: LoadingState = {
  loading: true,
}

export const loading = (
  state = Immutable.fromJS(initialState),
  action: AnyAction,
): string => {
  switch (action.type) {
    case 'SET_LOADING':
      return state.setIn(['loading'], action.value)
    default:
      return state
  }
}
