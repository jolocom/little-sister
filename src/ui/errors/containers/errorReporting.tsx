import React, { useState } from 'react'
import { connect } from 'react-redux'
import { NavigationScreenProp, NavigationState } from 'react-navigation'
import { AppError, reportError } from '../../../lib/errors'
import { ThunkDispatch } from '../../../store'
import { routeList } from '../../../routeList'
import { navigationActions } from '../../../actions'
import { Emoji, EmojiSection } from '../components/emojiSection'
import { JolocomButton, Wrapper } from '../../structure'
import { ScrollView } from 'react-native'
import { ChooseIssueSection } from '../components/chooseIssueSection'
import { DescriptionSection } from '../components/descriptionSection'
import { ContactSection } from '../components/contactSection'
import I18n from '../../../locales/i18n'
import strings from '../../../locales/strings'
import { SectionWrapper } from '../components/sectionWrapper'
import { NavigationSection } from '../components/navigationSection'

interface PaymentNavigationParams {
  error?: AppError | Error
  previousScreen: routeList
}

interface Props extends ReturnType<typeof mapDispatchToProps> {
  navigation: NavigationScreenProp<NavigationState, PaymentNavigationParams>
}

export enum Inputs {
  None,
  Dropdown,
  Description,
  Contact,
}

const ErrorReportingContainer = (props: Props) => {
  const { navigateToScreen, navigation } = props

  const [pickedIssue, setIssue] = useState('')
  const [description, setDescription] = useState('')
  const [contact, setContact] = useState('')
  const [currentInput, setInput] = useState(Inputs.None)
  const [toggleState, setToggle] = useState(false)
  const [selectedEmoji, setEmoji] = useState(Emoji.Empty)

  // NOTE send previous screen as props, since not all
  // error reports will have error objects

  // const { previousScreen } = navigation.state.params
  // const isBackButton = previousScreen === routeList.Settings
  const isBackButton = false

  // NOTE error reports can show up without an error,
  // hence a navigation prop (navigateTo) would be better than in the error object
  const navigateBack = () => {
    const error = props.navigation?.state?.params?.error
    const previousScreen = props.navigation?.state.params?.previousScreen
    if (error instanceof AppError) {
      navigateToScreen(error.navigateTo)
    } else {
      navigateToScreen(previousScreen || routeList.AppInit)
    }
  }

  const onSubmitReport = () => {
    const error = props.navigation?.state?.params?.error

    const userReport = {
      userError: pickedIssue,
      userDescription: description,
      userContact: contact,
      sendPrivateData: toggleState,
    }

    if (navigation) {
      reportError({ ...userReport, error })
      navigateBack()
    }
  }

  return (
    <Wrapper secondaryDark>
      <NavigationSection
        onNavigation={navigateBack}
        isBackButton={isBackButton}
      />
      <ScrollView overScrollMode="never">
        <SectionWrapper
          title={I18n.t(strings.TELL_US_THE_PROBLEM)}
          style={{ marginTop: 14 }}>
          <ChooseIssueSection
            currentInput={currentInput}
            pickedIssue={pickedIssue}
            onIssuePick={setIssue}
            setInput={setInput}
          />
        </SectionWrapper>
        <SectionWrapper title={I18n.t(strings.CAN_YOU_BE_MORE_SPECIFIC)}>
          <DescriptionSection
            currentInput={currentInput}
            setInput={setInput}
            setDescription={setDescription}
            toggleState={toggleState}
            setToggle={setToggle}
            description={description}
          />
        </SectionWrapper>
        <SectionWrapper title={I18n.t(strings.WANT_TO_GET_IN_TOUCH)}>
          <ContactSection
            onContactInput={setContact}
            currentInput={currentInput}
            setInput={setInput}
            contactValue={contact}
          />
        </SectionWrapper>
        <SectionWrapper title={I18n.t(strings.RATE_THE_ISSUE)}>
          <EmojiSection selectedEmoji={selectedEmoji} setEmoji={setEmoji} />
        </SectionWrapper>
        <JolocomButton
          onPress={onSubmitReport}
          containerStyle={{
            marginTop: 45,
            marginBottom: 66,
            marginHorizontal: 20,
            height: 56,
          }}
          text={I18n.t(strings.SUBMIT_REPORT)}
        />
      </ScrollView>
    </Wrapper>
  )
}

const mapDispatchToProps = (dispatch: ThunkDispatch) => ({
  navigateToScreen: (screen: routeList) =>
    dispatch(navigationActions.navigate({ routeName: screen })),
})

export const ErrorReporting = connect(
  null,
  mapDispatchToProps,
)(ErrorReportingContainer)
