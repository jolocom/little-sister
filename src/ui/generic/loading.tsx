import React from 'react'
import { Container, CenteredText } from 'src/ui/structure/'
import { Dimensions, Image, StyleSheet } from 'react-native'
import { JolocomTheme } from 'src/styles/jolocom-theme'

const image = require('src/resources/img/splashScreen.png')

interface Props {}

const styles = StyleSheet.create({
  loadingContainer: {
    backgroundColor: '#05050d',
  },
  loadingText: {
    position: 'absolute',
    bottom: '5%',
    fontSize: 12,
    fontFamily: JolocomTheme.contentFontFamily,
    color: JolocomTheme.primaryColorSand,
    opacity: 0.7,
  },
})

export const LoadingScreen: React.SFC<Props> = props => {
  const viewWidth: number = Dimensions.get('window').width
  const viewHeight: number = Dimensions.get('window').height

  return (
    <Container style={styles.loadingContainer}>
      <Image
        source={image}
        style={{
          bottom: '15%',
          width: viewWidth,
          height: viewHeight / 2,
        }}
      />
      <CenteredText style={styles.loadingText} msg={'POWERED BY JOLOCOM'} />
    </Container>
  )
}
