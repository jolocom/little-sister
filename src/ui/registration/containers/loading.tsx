import * as React from 'react'
import { connect } from 'react-redux'
import { View, StyleSheet } from 'react-native'
import Icon from 'react-native-vector-icons/FontAwesome'
import { registrationActions } from 'src/actions'
import { RootState } from 'src/reducers/'
import Immutable from 'immutable'
import { Container, CenteredText, Block } from 'src/ui/structure/'
import { JolocomTheme } from 'src/styles/jolocom-theme'
import I18n from 'src/locales/i18n'
const loaders = require('react-native-indicator')

export interface ConnectProps {
  loadingStage: number,
  loadingStages: string[]
  createIdentity: (encodedEntropy: string) => void
}

interface Props extends ConnectProps {
}

export interface State {
}

const styles = StyleSheet.create({
  label: {
    alignSelf: 'flex-start',
    marginBottom: '10%'
  },
  loadingMsg: {
    alignSelf: 'flex-end',
    marginBottom: '-10%'
  },
  container: {
    backgroundColor: JolocomTheme.primaryColorBlack,
    height: '100%'
  },
  dotsContainer: {
    alignItems: 'center',
    flexDirection: 'row',
    marginBottom: 20,
  },
  dotActive: {
    marginRight: 8,
    marginLeft: 8,
    color: JolocomTheme.primaryColorSand
  },
  dotInactive: {
    marginRight: 5,
    marginLeft: 5,
    color: JolocomTheme.primaryColorGrey
  },
  text: {
    color: JolocomTheme.primaryColorSand,
    fontSize: 20,
    fontFamily: JolocomTheme.contentFontFamily
  },
  smallText: {
    color: JolocomTheme.primaryColorSand,
    fontSize: 14,
    fontFamily: JolocomTheme.contentFontFamily
  }
})

// TODO SFC
export class LoadingContainer extends React.Component<Props, State> {
  render() {
    const {loadingStage, loadingStages} = this.props;
    return (
      <Container style={styles.container} >
        <Block style={styles.label} >
          <CenteredText style={styles.text} msg={ I18n.t('Give us a few moments') } />
          <CenteredText style={styles.text} msg={ I18n.t('to set up your identity') } />
        </Block>
        <Block>
          <loaders.RippleLoader size={80} strokeWidth={4} color={JolocomTheme.spinnerColor} />
        </Block>
        <Block style={styles.loadingMsg}>
          <View style={styles.dotsContainer}>
            {loadingStages.map((stage, key) => {
              return (
                <Icon
                  name='circle'
                  size={ key <= loadingStage ? 15 : 10 }
                  style={ key <= loadingStage ? styles.dotActive : styles.dotInactive }
                  key={ key }
                />
              )
            }) }
          </View>
          <View>
            <CenteredText style={styles.smallText} msg={loadingStages[loadingStage]} />
          </View>
        </Block>
      </Container>
    )
  }
}

const mapStateToProps = (state: RootState) => {
  const loading = Immutable.fromJS(state.registration.loading)
  return {
    loadingStage: loading.get('loadingStage'),
    loadingStages: loading.get('loadingStages'),
  }
}

const mapDispatchToProps = (dispatch: Function) => {
  return {
    createIdentity: (entropy: string) =>
      dispatch(registrationActions.createIdentity(entropy))
  }
}

export const Loading = connect(mapStateToProps, mapDispatchToProps)(LoadingContainer)
