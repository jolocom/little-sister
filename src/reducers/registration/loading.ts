import { AnyAction } from 'redux'
import Immutable from 'immutable'
import { LoadingState } from 'src/reducers/registration/'

const initialState : LoadingState = {
  loadingMsg: 'Loading'
}

export const loading = (state = Immutable.fromJS(initialState), action: AnyAction): string => {
  switch (action.type) {
    case 'SET_LOADING_MSG':
      return state.setIn(['loadingMsg'], action.value)
    default:
      return state
  }
}
