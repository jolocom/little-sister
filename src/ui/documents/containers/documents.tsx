import React from 'react'
import { connect } from 'react-redux'
import { Animated, ScrollView, StyleSheet, Text, View } from 'react-native'

import { openDocumentDetails } from 'src/actions/documents'
import { DecoratedClaims } from 'src/reducers/account'
import { RootState } from 'src/reducers'
import { getDocumentClaims } from 'src/utils/filterDocuments'
import { ThunkDispatch } from 'src/store'

import I18n from 'src/locales/i18n'
import { filters } from 'src/lib/filterDecoratedClaims'

import { DocumentsCarousel } from '../components/documentsCarousel'
import { DocumentsList } from '../components/documentsList'
import { DocumentViewToggle } from '../components/documentViewToggle'
import strings from '../../../locales/strings'
import { Typography, Colors } from 'src/styles'
import { Wrapper } from '../../structure'

interface Props
  extends ReturnType<typeof mapDispatchToProps>,
    ReturnType<typeof mapStateToProps> {}

interface State {
  activeDocumentIndex: number
  showingValid: boolean
}

const styles = StyleSheet.create({
  emptyDocumentsText: {
    ...Typography.mainText,
    ...Typography.centeredText,
    color: Colors.greyLight,
    paddingHorizontal: '5%',
  },
})

export class DocumentsContainer extends React.Component<Props, State> {
  public ScrollViewRef: ScrollView | null = null

  public state = {
    activeDocumentIndex: 0,
    showingValid: true,
  }

  private scrollToTop(): void {
    if (this.ScrollViewRef) {
      this.ScrollViewRef.scrollTo({ x: 0, y: 0, animated: true })
    }
  }

  private onActiveIndexChange(index: number): void {
    this.setState({ activeDocumentIndex: index })
    this.scrollToTop()
  }

  private handleToggle = () => {
    this.setState((prevState: State) => ({
      showingValid: !prevState.showingValid,
      activeDocumentIndex: 0,
    }))
    this.scrollToTop()
  }

  public render(): JSX.Element {
    const { openDocumentDetails, decoratedCredentials } = this.props
    const documents = getDocumentClaims(decoratedCredentials['Other'])
    const docFilter = this.state.showingValid
      ? filters.filterByValid
      : filters.filterByExpired
    const displayedDocs = docFilter(documents)
    const isEmpty = displayedDocs.length === 0
    const otherIsEmpty = displayedDocs.length === documents.length

    if (isEmpty) {
      return (
        <Wrapper centered>
          <Text style={styles.emptyDocumentsText}>
            {I18n.t(strings.NO_DOCUMENTS_TO_SEE_HERE) + '...'}
          </Text>
        </Wrapper>
      )
    }

    return (
      <Wrapper>
        <Animated.View style={{ flex: 1 }}>
          {!otherIsEmpty && (
            <DocumentViewToggle
              showingValid={this.state.showingValid}
              onTouch={this.handleToggle}
            />
          )}
          <ScrollView
            // to scroll to top upon changing card
            ref={ref => (this.ScrollViewRef = ref)}
            contentContainerStyle={
              isEmpty ? { flex: 1 } : { paddingBottom: 110 }
            }
            scrollEnabled={!isEmpty}
          >
            {this.state.showingValid ? (
              <DocumentsCarousel
                documents={displayedDocs}
                activeIndex={this.state.activeDocumentIndex}
                onActiveIndexChange={this.onActiveIndexChange.bind(this)}
              />
            ) : (
              <DocumentsList
                documents={displayedDocs}
                onDocumentPress={openDocumentDetails}
              />
            )}
          </ScrollView>
        </Animated.View>
      </Wrapper>
    )
  }
}

const mapStateToProps = ({
  account: {
    claims: { decoratedCredentials },
  },
}: RootState) => ({
  decoratedCredentials,
})

const mapDispatchToProps = (dispatch: ThunkDispatch) => ({
  openDocumentDetails: (doc: DecoratedClaims) =>
    dispatch(openDocumentDetails(doc)),
})

export const Documents = connect(
  mapStateToProps,
  mapDispatchToProps,
)(DocumentsContainer)
