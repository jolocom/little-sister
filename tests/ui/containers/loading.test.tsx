import React from 'react'
import { LoadingContainer } from 'src/ui/registration/containers/loading'
import { shallow } from 'enzyme'

describe('loading component', ()=> {
  it('matches the snapshot on render', () => {
    const props = {
      loadingStage: 0,
      loadingStages: ['Stage 1', 'Stage 2', 'Stage 3', ],
    }

    const rendered = shallow(<LoadingContainer {...props}/>)
    expect(rendered).toMatchSnapshot()
  })
})
