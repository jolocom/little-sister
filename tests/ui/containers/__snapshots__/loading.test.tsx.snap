// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loading component matches the snapshot on render 1`] = `
<Container
  style={
    Object {
      "backgroundColor": "black",
      "height": "100%",
    }
  }
>
  <Block
    style={
      Object {
        "alignSelf": "flex-start",
        "marginBottom": "10%",
      }
    }
  >
    <CenteredText
      msg="Give us a few moments"
      style={
        Object {
          "color": "#ffefdf",
          "fontFamily": "TT Commons",
          "fontSize": 20,
        }
      }
    />
    <CenteredText
      msg="to set up your identity"
      style={
        Object {
          "color": "#ffefdf",
          "fontFamily": "TT Commons",
          "fontSize": 20,
        }
      }
    />
  </Block>
  <Block>
    <RippleLoader
      color="#89ffdebc"
      size={80}
      strokeWidth={4}
    />
  </Block>
  <Block
    style={
      Object {
        "alignSelf": "flex-end",
        "marginBottom": "-10%",
      }
    }
  >
    <Component
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 20,
        }
      }
    >
      <Icon
        allowFontScaling={false}
        key="0"
        name="circle"
        size={15}
        style={
          Object {
            "color": "#ffefdf",
            "marginLeft": 8,
            "marginRight": 8,
          }
        }
      />
      <Icon
        allowFontScaling={false}
        key="1"
        name="circle"
        size={10}
        style={
          Object {
            "color": "#fafafa",
            "marginLeft": 5,
            "marginRight": 5,
          }
        }
      />
      <Icon
        allowFontScaling={false}
        key="2"
        name="circle"
        size={10}
        style={
          Object {
            "color": "#fafafa",
            "marginLeft": 5,
            "marginRight": 5,
          }
        }
      />
    </Component>
    <Component>
      <CenteredText
        msg="Stage 1"
        style={
          Object {
            "color": "#ffefdf",
            "fontFamily": "TT Commons",
            "fontSize": 14,
          }
        }
      />
    </Component>
  </Block>
</Container>
`;
